File: apis/elysium-core-api/src/main/java/com/seventh_root/elysium/core/ElysiumCore.java
package com.seventh_root.elysium.core;

import com.seventh_root.elysium.core.database.Database;
import com.seventh_root.elysium.core.service.ServiceManager;

import java.util.logging.Logger;

public class ElysiumCore {

    private final Logger logger;
    private final ServiceManager serviceManager;
    private final Database database;

    public ElysiumCore(Logger logger, Database database) {
        this.logger = logger;
        serviceManager = new ServiceManager();
        this.database = database;
    }

    public Logger getLogger() {
        return logger;
    }

    public ServiceManager getServiceManager() {
        return serviceManager;
    }

    public Database getDatabase() {
        return database;
    }

}

File: elysium-core/src/main/kotlin/com/seventh_root/elysium/core/ElysiumCore.kt
package com.seventh_root.elysium.core

import com.seventh_root.elysium.core.database.Database
import com.seventh_root.elysium.core.service.ServiceManager

import java.util.logging.Logger

class ElysiumCore(val logger: Logger, val database: Database) {

    val serviceManager: ServiceManager

    init {
        serviceManager = ServiceManager()
    }

}
